### PR TITLE
Pass listify_dataset as a hyperparameter through the shell

### PR DESCRIPTION
**Pull request by dcmaddix**
- Pass `listify_dataset` as a boolean field in `Hyperparameters` dictionary through the shell to determine whether or not to cache the dataset
- If `listify_dataset = True`, we call `ListDataset` otherwise the dataset type is `FileDataset`
- Added corresponding test: `test_listify_dataset`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/shell/sagemaker/__init__.py [hunk 1]
```diff
@@ -131,8 +134,19 @@ def _load_datasets(
...
+    )
+    dataset_dict = {}
+    for name in DATASET_NAMES:
+        if name in channels:
+            file_dataset = FileDataset(channels[name], freq)
+            dataset_dict[name] = (
+                ListDataset(file_dataset, freq)
+                if listify_dataset
+                else file_dataset
+            )
```
Comment (AaronSpieler) [inline src/gluonts/shell/sagemaker/__init__.py:150]: I would add the try catch logic for catching OOM due to the conversion of FileDataset to ListDataset here.

File: src/gluonts/shell/sagemaker/__init__.py [hunk 2]
```diff
@@ -131,8 +134,19 @@ def _load_datasets(
...
-    return {
-        name: FileDataset(channels[name], freq)
-        for name in DATASET_NAMES
-        if name in channels
-    }
+    listify_dataset = (
+        strtobool(hyperparameters["listify_dataset"])
+        if "listify_dataset" in hyperparameters.keys()
+        else False
+    )
```
Comment (jaheba) [inline src/gluonts/shell/sagemaker/__init__.py:141]:
```suggestion
    listify_dataset = strtobool(hyperparameters.get("listify_dataset", "no"))
```

File: test/shell/test_shell.py [hunk 3]
```diff
@@ -112,6 +125,7 @@ def test_train_shell(train_env: TrainEnv, caplog) -> None:
             assert line.endswith("270.0")
 
 
+@pytest.mark.parametrize("listify_dataset", [True])
```
Comment (jaheba) [inline test/shell/test_shell.py:128]: I guess we would want to test `True` and `False`.

Comment (dcmaddix) [reply test/shell/test_shell.py:128]: I can add the `False` cases to all the tests as well, but I added a separate test `test_listify_dataset` that tests both `True` and `False`.

File: src/gluonts/shell/sagemaker/__init__.py (hunk 2 above)
Comment (dcmaddix) [reply src/gluonts/shell/sagemaker/__init__.py:141]: Great, will update thanks!

**Pull request merged**